collision/rtree.go

```diff
@@ -359,12 +358,12 @@ func (tree *Rtree) condenseTree(n *node) error {
 					entries = append(entries, e)
 				}
 			}
-			if len(n.parent.entries) == len(entries) {
-				// This suggests the tree is malformed, as the child has a
-				// reference to a parent that is not aware of them as a child
-				// in our experience we've never seen this error occur.
-				return fmt.Errorf("Failed to remove entry from parent")
-			}
+			// if len(n.parent.entries) == len(entries) {
```

Comment:
Is the idea here to leave the dead code just in case?

Reply:
Yeah, its a part of the engine that rarely ever gets touched and that differs from other rtree implementations, so I think keeping the check with the comment explaining why it is commented out is valid.